src/ophyd_async/core/_detector.py

```diff
@@ -73,32 +77,36 @@ class DetectorControl(ABC):
     arming and disarming a detector
     """
 
+    _arm_status = AsyncStatus
```

Comment:
```suggestion
    _arm_status: AsyncStatus | None
```

This does require that all subclasses of DetectorControl have this field, and that it is this type. As arm is abstract, and we're leaving implementation to the implementations I'd avoid having any required fields like this.